crates/wast/src/ast/types.rs

```diff
@@ -66,6 +69,244 @@ impl<'a> Peek for ValType<'a> {
     }
 }
 
+/// An interface-types type.
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum InterType {
+    Bool,
+    S8,
+    U8,
+    S16,
+    U16,
+    S32,
+    U32,
+    S64,
+    U64,
+    Float32,
+    Float64,
+    Char,
+    String,
+    Record(Record),
+    Variant(Variant),
+    List(List),
+    Tuple(Tuple),
+    Flags(Flags),
+    Enum(Enum),
+    Union(Union),
+    Optional(Optional),
+    Expected(Expected),
+    Named(Named),
+}
+
+impl<'a> Parse<'a> for InterType {
+    fn parse(parser: Parser<'a>) -> Result<Self> {
+        let mut l = parser.lookahead1();
+        if l.peek::<kw::r#bool>() {
+            parser.parse::<kw::r#bool>()?;
+            Ok(InterType::Bool)
+        } else if l.peek::<kw::r#bool>() {
+            parser.parse::<kw::r#bool>()?;
+            Ok(InterType::Bool)
+        } else if l.peek::<kw::s8>() {
+            parser.parse::<kw::s8>()?;
+            Ok(InterType::S8)
+        } else if l.peek::<kw::r#u8>() {
+            parser.parse::<kw::r#u8>()?;
+            Ok(InterType::U8)
+        } else if l.peek::<kw::s16>() {
+            parser.parse::<kw::s16>()?;
+            Ok(InterType::S16)
+        } else if l.peek::<kw::r#u16>() {
+            parser.parse::<kw::r#u16>()?;
+            Ok(InterType::U16)
+        } else if l.peek::<kw::s32>() {
+            parser.parse::<kw::s32>()?;
+            Ok(InterType::S32)
+        } else if l.peek::<kw::r#u32>() {
+            parser.parse::<kw::r#u32>()?;
+            Ok(InterType::U32)
+        } else if l.peek::<kw::s64>() {
+            parser.parse::<kw::s64>()?;
+            Ok(InterType::S64)
+        } else if l.peek::<kw::r#u64>() {
+            parser.parse::<kw::r#u64>()?;
+            Ok(InterType::U64)
+        } else if l.peek::<kw::float32>() {
+            parser.parse::<kw::float32>()?;
+            Ok(InterType::Float32)
+        } else if l.peek::<kw::float64>() {
+            parser.parse::<kw::float64>()?;
+            Ok(InterType::Float64)
+        } else if l.peek::<kw::r#char>() {
+            parser.parse::<kw::r#char>()?;
+            Ok(InterType::Char)
+        } else if l.peek::<kw::string>() {
+            parser.parse::<kw::string>()?;
+            Ok(InterType::String)
+        } else if l.peek::<kw::record>() {
+            let record = parser.parse::<Record>()?;
+            Ok(InterType::Record(record))
+        } else if l.peek::<kw::variant>() {
+            let variant = parser.parse::<Variant>()?;
+            Ok(InterType::Variant(variant))
+        } else if l.peek::<kw::tuple>() {
```

Comment:
The parsing here may end up being a little different (for everything else here that isn't just a plain keyword as well) because I suspect we'd want this represented as `(variant ...)` or `(tuple ...)` surrounded in parens.